packages/@lwc/engine-core/src/libs/signal-tracker/index.ts

```diff
@@ -5,8 +5,8 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
 import { isFalse, isFunction, isUndefined } from '@lwc/shared';
-import { Signal } from '@lwc/signals';
 import { logWarnOnce } from '../../shared/logger';
+import type { Signal } from '@lwc/signals';
```

Comment:
This type needs to be re-exported in `SignalTracker` as part of the function definition types.